flopy4/array.py

```diff
@@ -203,6 +203,7 @@ def __init__(
         tagged=False,
         reader=MFReader.urword,
         default_value=None,
+        extpath=None,
```

Comment:
imo this could be public, an attr or property of type `Optional[Path]` that has a value if the array is external otherwise `None`? I'd also vote to call it `path`